### ISSUE TITLE
headers field should be case insensitive

### ISSUE THREAD

**Issue by KaiPetzke** [mrtazz/restclient-cpp#157]
### Expected behaviour

The following code snippet:
    RestClient::Connection* conn = new RestClient::Connection("https://httpbin.org");
    RestClient::Response& r = conn->get("/get");
    for(auto& name : { "date", "Date" }) {
        try {
            std::string& value = r.headers.at(name);
            std::cout << name << ": \"" << value << "\"" << std::endl;
        } catch(std::out_of_range& oor) {
            std::cout << name << ": not found" << std::endl;
        }
    }

should yield a value similiar to the following:

date: "Tue, 11 Aug 2020 19:52:17 GMT"
Date: "Tue, 11 Aug 2020 19:52:17 GMT"

### Actual behaviour

The result is as follows:

date: not found
Date: "Tue, 11 Aug 2020 19:52:17 GMT"

This is an error, as according to RFC 2616: "Field names are case-insensitive". And yes, unfortunately, there are servers, that change the case of their response header fields at will.

### Environment and debugging details

This happens with any C compiler and libcurl

### Suggested fix:

Use a custom compare function to make the std::map in RestClient::HeaderFields case-insensitive. I have attached a file with the suggested change. Yes, it uses std::tolower(), which unfortunately is locale dependant, for conversion to lowercase. So depending on language settings, tolower() might convert, say, Ç to ç or not. If that is considered an issue, an alternative would be to explicitely only map characters in the range 'A' to 'Z' to 'a' to 'z', and use this mapping function instead of std::tolower():
    unsigned char ascii_tolower(const unsigned char c) {
        return (c >= 'A' && c <= 'Z') ? ((c - 'A') + 'a') : c;
    }

[restclient.diff.txt](https://github.com/mrtazz/restclient-cpp/files/5059211/restclient.diff.txt)

**Comment by KaiPetzke** [comment]
Yeah, I didn't read the contribution guidelines. I am busy this week, will try to add a formal fix next week.

@edwinpjacques What do you suggest about my comments below the headline "suggested fix"? Do you prefer `std::tolower()` or the locale independant `ascii_tolower()`?

**Comment by edwinpjacques** [comment]
std::tolower looks good. https://en.cppreference.com/w/cpp/string/byte/tolower

**Comment by KaiPetzke** [comment]
I am still busy, will look at all these issues early next year.

**Comment by Mirza585** [comment]
Shall I fix the issue?

**Comment by Ayushman2004** [comment]
Opened a PR #203
